filelog.go

```diff
@@ -128,12 +128,21 @@ func NewFileLogWriter(fname string, rotate bool) *FileLogWriter {
 			}
 		}()
 
+		// Failure counters
+		var rotationFailures uint64
+		var writeFailures uint64
+
 		for {
 			select {
 			case <-w.rot:
 				if err := w.handleRotate(time.Now()); err != nil {
 					fmt.Fprintf(os.Stderr, "FileLogWriter(%q): %s\n", w.filename, err)
-					return
+					rotationFailures += 1
+				} else {
+					if rotationFailures != 0 {
+						_, logErr := fmt.Fprintf(os.Stderr, "FileLogWriter(%q): Ignored %d previous rotation attempts due to rotation failure\n", w.filename, rotationFailures)
+						if logErr == nil { rotationFailures = 0 }
```

Comment:
Should this be resetting `writeFailures`? Also the `logErr` variable name is a bit misleading as it's "error writing to stderr (which I attempted because I couldn't write to the log)" and not "error writing to the log"